app/src/main/java/org/listenbrainz/android/ui/screens/yim23/Yim23StatsGraphScreen.kt

```diff
@@ -0,0 +1,125 @@
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import org.listenbrainz.android.model.yimdata.Yim23Screens
+import org.listenbrainz.android.ui.components.Yim23Footer
+import org.listenbrainz.android.ui.components.Yim23Header
+import org.listenbrainz.android.ui.theme.LocalYimPaddings
+import org.listenbrainz.android.ui.theme.Yim23Theme
+import org.listenbrainz.android.ui.theme.YimPaddings
+import org.listenbrainz.android.ui.theme.offWhite
+import org.listenbrainz.android.viewmodel.Yim23ViewModel
+
+@Composable
+fun Yim23StatsGraphScreen (
+    viewModel: Yim23ViewModel,
+    navController: NavController
+) {
+    val username by viewModel.getUsernameFlow().collectAsState(initial = "")
+    val mostListenedYear = viewModel.getMostListenedYear()!!
+    Yim23Theme(themeType = viewModel.themeType.value) {
+        Column (modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.onBackground) , verticalArrangement = Arrangement.SpaceBetween) {
+            Yim23Header(username = username, navController = navController)
+            Spacer(modifier = Modifier.padding(top = 11.dp))
+            Text("Most of the songs I listened to were from ${mostListenedYear.key} (${mostListenedYear.value} songs)" , style = MaterialTheme.typography.bodyLarge , color = MaterialTheme.colorScheme.background , textAlign = TextAlign.Center)
+            Spacer(modifier = Modifier.padding(top = 21.dp))
+            Yim23Graph(viewModel = viewModel)
+            Yim23Footer(footerText = "MY STATS", isUsername = true, navController = navController, downScreen = Yim23Screens.YimPlaylistsTitleScreen)
+        }
+    }
+}
+
+
+@Composable
+private fun Yim23Graph (viewModel: Yim23ViewModel , paddings: YimPaddings = LocalYimPaddings.current) {
+    val listState = rememberLazyListState()
+    val graphState = rememberLazyListState()
+
+    Box (modifier = Modifier
+        .fillMaxWidth()
+        .padding(start = 11.dp, end = 11.dp)
+        .height(250.dp)
+        .clip(RoundedCornerShape(10.dp))
+        .background(Color(0xFFe0e5de)) , contentAlignment = Alignment.TopCenter) {
+        LazyRow (state = graphState) {
+            items(viewModel.getYearListens().toList()) { item ->
+                val height = (item.second * 250) / (viewModel.getMostListenedYear()!!.value)
```

Comment:
I would suggest using graph library here, google some suitable library else, make a component of this graph and make sure its similar to website.
Also wrap these calculations in remember.